Mathlib/Analysis/NormedSpace/MStructure.lean

```diff
@@ -311,3 +312,78 @@ instance Subtype.BooleanAlgebra [FaithfulSMul M X] :
     sdiff_eq := fun P Q => Subtype.ext <| by rw [coe_sdiff, ← coe_compl, coe_inf] }
 
 end IsLprojection
+
+section WithL1
+
+open ENNReal
+
+variable (p : ℝ≥0∞) (α β : Type*)
+
+variable {p α β}
+variable [NormedAddCommGroup α] [NormedAddCommGroup β]
+
+/-- Projection on `WithLp p (α × β)` with range `α` and kernel `β` -/
+def P1 : AddMonoid.End (WithLp p (α × β)) := (AddMonoidHom.inl α β).comp (AddMonoidHom.fst α β)
+
+/-- Projection on `WithLp p (α × β)` with range `β` and kernel `α` -/
+def P2 : AddMonoid.End (WithLp p (α × β)) := (AddMonoidHom.inr α β).comp (AddMonoidHom.snd α β)
+
+lemma P1_apply (x : WithLp p (α × β)) : P1 x = (x.1, 0) := rfl
+
+lemma P2_apply (x : WithLp p (α × β)) : P2 x = (0, x.2) := rfl
+
+lemma P1_compl : (1 : AddMonoid.End (WithLp p (α × β))) - P1 = P2 := AddMonoidHom.ext
+  fun x => by
+    rw [AddMonoidHom.sub_apply, P1_apply, P2_apply, AddMonoid.End.coe_one, id_eq,
+      sub_eq_iff_eq_add, add_comm, WithLp.mk_add_mk, zero_add, add_zero]; rfl
+
+theorem prod_norm_eq_sup (x : WithLp ∞ (α × β)) : ‖x‖ = ‖P1 x‖ ⊔ ‖(1 - P1) x‖ := by
+  rw [WithLp.prod_norm_eq_sup x, P1_apply, P1_compl, P2_apply]
+  apply congr_arg₂
+  · simp only [WithLp.prod_norm_eq_sup, norm_zero, norm_nonneg, sup_of_le_left]
+  · simp only [WithLp.prod_norm_eq_sup, norm_zero, norm_nonneg, sup_of_le_right]
+
+lemma prod_norm_eq_add_P1 (hp : 0 < p.toReal) (x : WithLp p (α × β)) :
+    ‖x‖ = (‖P1 x‖ ^ p.toReal + ‖(1 - P1) x‖ ^ p.toReal) ^ (1 / p.toReal) := by
+  rw [WithLp.prod_norm_eq_add hp x, P1_apply, P1_compl, P2_apply]
+  apply congr_arg₂
+  · have ez : p.toReal ≠ 0 := (ne_of_lt hp).symm
+    have e1 : (0 : ℝ) ^ p.toReal = (0 : NNReal) := by
+      rw [← (NNReal.zero_rpow ez)]
+      rfl
+    apply congr_arg₂
+    · apply congr_arg₂
+      · rw [WithLp.prod_norm_eq_add hp (x.1, 0)]
+        simp only [norm_zero]
+        rw [e1]
+        simp only [NNReal.coe_zero, add_zero, norm_nonneg]
+        convert (NNReal.rpow_inv_rpow_self ez ⟨‖x.1‖,norm_nonneg x.1⟩)
+        simp_all only [ne_eq, not_false_eq_true, Real.zero_rpow, NNReal.coe_zero, one_div,
+          norm_nonneg, Real.rpow_rpow_inv, NNReal.rpow_rpow_inv]
+      · rfl
+    · apply congr_arg₂
+      · rw [WithLp.prod_norm_eq_add hp (0,x.2)]
+        simp only [norm_zero]
+        rw [e1]
+        simp only [NNReal.coe_zero, add_zero, norm_nonneg]
+        convert (NNReal.rpow_inv_rpow_self ez ⟨‖x.2‖,norm_nonneg x.2⟩)
+        simp_all only [ne_eq, not_false_eq_true, Real.zero_rpow, NNReal.coe_zero, zero_add, one_div,
+          norm_nonneg, Real.rpow_rpow_inv, NNReal.rpow_rpow_inv]
+      · rfl
+  · rfl
+
+lemma WithLp.prod_norm_eq_of_1 (x : WithLp 1 (α × β)) :
```

Comment:
I've split this to #20383, along with a bunch of related lemmas.

Reply:
Thanks!

Reply:
Now that we have a simple proof of `prod_norm_eq_add_idemFst`, I've just used that result rather than treat `p = 1` as a special case.